addons/l10n_it_edi/tests/test_ir_mail_server.py

```diff
@@ -164,3 +165,22 @@ def test_ricevuta_consegna(self):
     def test_decorrenza_termini(self):
         """ Test a receipt adapted from https://www.fatturapa.gov.it/export/documenti/messaggi/v1.0/IT01234567890_11111_DT_001.xml """
         self._test_receipt('DT', 'delivered', 'delivered_expired')
+
+    # -----------------------------
+    #
+    # Mail server
+    #
+    # -----------------------------
+
+    def test_server_archived_usage_protection(self):
+        """ Test the protection against using archived server (servers used cannot be archived) """
+        IrMailServer = self.env['ir.mail_server']
+        server = IrMailServer.create({
+            'name': 'Server',
+            'smtp_host': 'archive-test.smtp.local',
+        })
+        self.company.l10n_it_mail_pec_server_id = server.id
+        with self.assertRaises(UserError, msg='Server cannot be archived because it is used'):
+            server.write({'active': False})
```

Comment:
```suggestion
            server.action_unarchive()
```

Reply:
Thanks, I didn't know that method, changed